lib/workers/repository/update/branch/index.ts

```diff
@@ -487,10 +488,11 @@ export async function processBranch(
 
     // compile commit message with body, which maybe needs changelogs
     if (config.commitBody) {
-      // TODO: defer fetching changelogs (#17020)
-      // if (config.fetchReleaseNotes && needsChangelogs(config, ['commitBody'])) {
-      //   await embedChangelog(config);
-      // }
+      if (config.fetchReleaseNotes && needsChangelogs(config, ['commitBody'])) {
+        // we only need first upgrade, the others are only needed on PR update
+        // we add it to first, so PR fetch can skip fetching for that update
+        await embedChangelog(config.upgrades[0]);
```

Comment:
Is `config.upgrades` guaranteed to have at least one element?

Reply:
yes, renovate will never get here if the branch hasn't any dep to update